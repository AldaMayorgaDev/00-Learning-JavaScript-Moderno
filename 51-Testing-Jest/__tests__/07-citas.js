import Citas from '../js/classes/Citas';

describe('Probar la clase de Citas', ()=>{
    const citas = new Citas();
    const id = Date.now();

    test('Test Agregar una nueva cita', () => { 

        const citaObj = {
            id,
            mascota: 'hokk',
            propietario: 'Aldahir',
            telefono: '888338212',
            fecha: '10-05-2023',
            hora:'15:03',
            sintomas: 'Solo duerme'
        }


        citas.agregarCita(citaObj);

        //PRueba
        expect(citas).toMatchSnapshot(); /*Esto va a crear un snapshot que va a tener la informacion y vamos a poder verificar que la informacion que le estamos pasando a la clase (en el bojeto citObj ) se estÃ© guardando correctamente.*/

     });

     test('Test de Actualizar cita', ()=>{
        const citaActualizada = {
            id,
            mascota: 'Nombre Actualizada',
            propietario: 'Aldahir',
            telefono: '888338212',
            fecha: '10-05-2023',
            hora:'15:03',
            sintomas: 'Solo duerme'
        }
        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
     });

     test('Test de Eliminar citas', ()=>{
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
     })
})