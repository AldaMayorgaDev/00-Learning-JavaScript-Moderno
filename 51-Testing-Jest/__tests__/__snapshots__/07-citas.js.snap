// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Test Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-05-2023",
      "hora": "15:03",
      "id": 1682991171062,
      "mascota": "hokk",
      "propietario": "Aldahir",
      "sintomas": "Solo duerme",
      "telefono": "888338212",
    },
  ],
}
`;

exports[`Probar la clase de Citas Test de Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-05-2023",
      "hora": "15:03",
      "id": 1682991171062,
      "mascota": "Nombre Actualizada",
      "propietario": "Aldahir",
      "sintomas": "Solo duerme",
      "telefono": "888338212",
    },
  ],
}
`;

exports[`Probar la clase de Citas Test de Eliminar citas 1`] = `
Citas {
  "citas": [],
}
`;
